01-Login/src/Auth/Auth.js

```diff
@@ -25,40 +33,52 @@ export default class Auth {
 
   handleAuthentication() {
     this.auth0.parseHash((err, authResult) => {
-      if (authResult && authResult.accessToken && authResult.idToken) {
-        this.setSession(authResult);
-        history.replace('/home');
-      } else if (err) {
-        history.replace('/home');
+      if (err) {
+        this.goTo('/home');
         console.log(err);
         alert(`Error: ${err.error}. Check the console for further details.`);
+      } else if (authResult && authResult.accessToken) {
+        this.storeToken(authResult);
+        this.goTo('/home');
       }
     });
   }
 
-  setSession(authResult) {
-    // Set the time that the access token will expire at
-    let expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
-    localStorage.setItem('access_token', authResult.accessToken);
-    localStorage.setItem('id_token', authResult.idToken);
-    localStorage.setItem('expires_at', expiresAt);
-    // navigate to the home route
-    history.replace('/home');
+  renewAuthentication() {
+    this.auth0.checkSession({}, (err, authResult) => {
+        if (err) {
+          this.logout();
+          console.log(err);
```

Comment:
When we call renewAuthentication() when the application starts and checkSession() fails, we should not give an error but set the application to 'unauthenticated' state. Check how it's done in the Angular sample that Dan did.